### PR TITLE
47 config front end

### PR DESCRIPTION
**Pull request by IssaHanou**
### Issue
Closes GH-47
This was branched of closes GH-39 due to menu already implemented there

## Contents
- adds page for entering config on front-end
- shows errors in box
    1. JSON errors shown first, one by one - because of golang json, can write manual key checker to show all at once GH-76
    2. Errors shown on incorrect formats/wrong strings shown second
    3. errors checking for matching id's etc shown third
- if there are no errors, user can select to use the entered config for the app

## TODO
- [x] testing the handler methods of the config don't work yet due to the loading of json from file and then sending it in json message
- [x] check that reset still works with correct config after loading

## Screen
big screen errors in config
image

small screen no errors
image

### CODE REVIEW

**Comment by GwennanS** [comment]
it does not remove old puzzles from the list

**Comment by GwennanS** [comment]
room_config is a little bit verneukt

File: back-end/src/sciler/config/readConfigTypes.go [hunk 1]
```diff
@@ -111,11 +111,12 @@ type Action struct {
 }
 
 // Execute is a method that performs the action
+// TODO test this
```
Comment (mrlomar) [inline back-end/src/sciler/config/readConfigTypes.go]: Is this todo done?

File: back-end/src/sciler/config/workingConfigTypes.go [hunk 2]
```diff
@@ -187,32 +188,35 @@ func compare(param1 interface{}, param2 interface{}, comparision string) bool {
...
 	default:
-		panic(fmt.Sprintf("cannot compare on: %s", comparision))
+		// This case is already handled to give error in checkConstraint
+		return false
 	}
 }
 
-func numericToFloat64(input interface{}) float64 {
+// NumericToFloat64 checks if numeric value is in or float64 and returns float64
+func NumericToFloat64(input interface{}) float64 {
```
Comment (mrlomar) [inline back-end/src/sciler/config/workingConfigTypes.go]: Why is this function public?

File: back-end/src/sciler/config/readConfigTypes.go (hunk 1 above)
Comment (IssaHanou) [reply back-end/src/sciler/config/readConfigTypes.go]: todo in GH-147

**Pull request merged**